sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdk.java

```diff
@@ -78,27 +78,27 @@ public BatchRecorder newMeasureBatchRecorder() {
 
   @Override
   public LabelSet createLabelSet(String k1, String v1) {
-    throw new UnsupportedOperationException("to be implemented");
+    return SdkLabelSet.builder().add(k1, v1).build();
   }
 
   @Override
   public LabelSet createLabelSet(String k1, String v1, String k2, String v2) {
-    throw new UnsupportedOperationException("to be implemented");
+    return SdkLabelSet.builder().add(k1, v1).add(k2, v2).build();
   }
 
   @Override
   public LabelSet createLabelSet(String k1, String v1, String k2, String v2, String k3, String v3) {
-    throw new UnsupportedOperationException("to be implemented");
+    return SdkLabelSet.builder().add(k1, v1).add(k2, v2).add(k3, v3).build();
   }
 
   @Override
   public LabelSet createLabelSet(
       String k1, String v1, String k2, String v2, String k3, String v3, String k4, String v4) {
-    throw new UnsupportedOperationException("to be implemented");
+    return SdkLabelSet.builder().add(k1, v1).add(k2, v2).add(k3, v3).add(k4, v4).build();
```

Comment:
Avoid extra allocation of the builder, probably you should avoid using the builder pattern for the labelSet

Reply:
huh. do you think labelset creation is likely to be on the hot path?

Reply:
definitely will, you need to create a LableSet almost every time you record a value.

Reply:
True. I wonder if we should ditch the LabelSet object allocation altogether and just use a standard Map, then.

Reply:
I guess the immutability is actually really important on a LabelSet, though, since it's effectively going to be used as a key to a map internally somewhere.

Reply:
I cleaned this up. I think we should change the API to just take varargs for labelset creation, rather than having only up to 4 keys allowed.

Reply:
#723 to track that change